app/controllers/users_controller.rb

```diff
@@ -0,0 +1,48 @@
+class UsersController < ApplicationController
+  before_action :set_user, only: [:edit, :update, :destroy]
+
+
+  def edit
+    if !current_user.is_admin
+      redirect_to user_path
+    end
+    @user
+  end
+
+  def update
+    respond_to do |format|
+      if @user.update(post_params)
+        @user.save
```

Comment:
no need to `save` a record after `update`. it is automatically persisted in database after `update`